include/deal.II/lac/sparse_matrix.h

```diff
@@ -1468,6 +1468,13 @@ class SparseMatrix : public virtual Subscriptor
               const bool  diagonal_first = true) const;
 
   /**
+   * Print the matrix to the given stream in the "Matrix Market" format.
+   */
+  template <class StreamType>
+  void print_matrix_market(StreamType &out,
+			   const number threshold = 1e-10) const;
```

Comment:
i think this should be `double`.  You probably won't specify threshold as a `std::compelx` number.